packages/hurumap-core/src/Scope/Scope.js

```diff
@@ -0,0 +1,224 @@
+import data from "./data";
+import signals from "./signals";
+import { createFiltersForGroups } from "./utils";
+
+export default function Scope({
+  primaryData,
+  metadata,
+  config,
+  secondaryData,
+  primaryParentData,
+  secondaryParentData,
+  chartType,
+  transform = [],
+  theme,
+  args,
+}) {
+  const { primary_group: primaryGroup, groups } = metadata;
+  const stackedField = config?.stacked_field;
+  const nestedFields = config?.nest_fields;
+
+  let groupBy = [primaryGroup];
+  if ((chartType === "stacked" || chartType === "line") && stackedField) {
+    groupBy = [primaryGroup, stackedField];
+  } else if (chartType === "treemap" && nestedFields) {
+    groupBy = nestedFields;
+  }
+
+  const { signals: filterSignals, filters } = createFiltersForGroups(groups);
+
+  return {
+    $schema: "https://vega.github.io/schema/vega/v5.json",
+    description: "A",
+    config: {
+      axis: {
+        labelColor: theme.palette.chart.text.primary,
+        labelFont: theme.typography.fontFamily,
+        labelFontSize: 11,
```

Comment:
@m453h much like the comment above, your suggestion wouldn't be possible. [This is](https://mui.com/material-ui/customization/default-theme/) how material UI theme object looks like. Unless your suggestion is to add custom variables specifically for chart.

Reply:
@m453h as mentioned during the daily scrum call, this can be added in the next iteration. I have created [this](https://github.com/CodeForAfrica/ui/issues/796) issue to track this.